services/friendbot/internal/friendbot.go

```diff
@@ -92,16 +94,31 @@ func (bot *Bot) checkSequenceRefresh() error {
 	return bot.refreshSequence()
 }
 
-func (bot *Bot) makeTx(destAddress string, amount string) (string, error) {
-	txn, err := b.Transaction(
-		b.SourceAccount{AddressOrSeed: bot.Secret},
-		b.Sequence{Sequence: bot.sequence + 1},
-		b.Network{Passphrase: bot.Network},
-		b.CreateAccount(
-			b.Destination{AddressOrSeed: destAddress},
-			b.NativeAmount{Amount: amount},
-		),
-	)
+func (bot *Bot) makeTx(destAddress string, amount string, isFundRequest bool) (string, error) {
+	var txn *b.TransactionBuilder
+	var err error
+
+	if (isFundRequest) {
+		txn, err = b.Transaction(
```

Comment:
you can move the common code outside of the condition sinc these are functional arguments:
```golang
ops := []build.TransactionMutator{
	b.SourceAccount{AddressOrSeed: bot.Secret},
	b.Sequence{Sequence: bot.sequence + 1},
	b.Network{Passphrase: bot.Network},
}

if (isFundRequest) {
	ops = append(ops, b.Payment(
		b.Destination{AddressOrSeed: destAddress},
		b.NativeAmount{Amount: amount})
} else {
	ops = b.CreateAccount(
		b.Destination{AddressOrSeed: destAddress},
		b.NativeAmount{Amount: amount})
}

// Also no need to define txn and err before using them. just use ":=" instead
txn, err := b.Transaction(ops...)
```

Reply:
see here for an additional example https://github.com/kinecosystem/stellar-load-testing/blob/cf28eace4211c992c81fdee5b5d82311a497529b/cmd/create/account.go#L78

Reply:
Fixed.